.ci_support/migrations/cuda120.yaml

```
@@ -0,0 +1,106 @@
+migrator_ts: 1682985063
+__migrator:
+  kind:
+    version
+  migration_number:
```

It looks like we are running into a known incompatibility issue with pybind11 + GCC 12 + NVCC 12

Newer versions of pybind11 or NVCC would fix the issue. However we have neither here. So this is a workaround to vendor the migrator and use GCC 11 throughout. Added a few suggestions related to this below

```suggestion
  # Vendor CUDA 12 migrator to use GCC 11.
  # This is needed to workaround a pybind + GCC 12 + nvcc 12 bug
  # xref: https://github.com/pybind/pybind11/issues/4606
  use_local: true
  migration_number:
```